tomcat-11.0.yaml

```diff
@@ -151,9 +151,33 @@ subpackages:
             export LD_LIBRARY_PATH="/usr/lib/tomcat-native"
 
             /usr/share/tomcat/bin/startup.sh
-            sleep 5
-
-            curl --silent --retry 3 --retry-connrefused http://localhost:8080/ | grep "Apache Tomcat" || {
+            echo "=== Waiting for Tomcat startup ==="
+            startup_success=false
```

Comment:
I have a suggestion.

This `startup_success` is never read, so you may as well drop it.

The wait loop could be a bit more compact, something like this:

```
timeout -k 1 30 sh -x -c 'while true ; do
  sleep 1
  if ! pgrep -f "[o]rg.apache.catalina"; then
      echo "tomcat proc not found yet" ; continue

  fi
  if ! netstat -ln 2>/dev/null | grep -q :8080 && ! ss -ln 2>/dev/null | grep -q :8080 ; then
      echo "port 8080 not open yet" ; continue
  fi
done'
```

Reply:
Love it thanks